csrc/pf_core.c

```diff
@@ -223,6 +223,10 @@ PForthDictionary pfCreateDictionary( cell_t HeaderSize, cell_t CodeSize )
 
     dic->dic_Flags |= PF_DICF_ALLOCATED_SEGMENTS;
 
+#ifdef PF_USE_STATIC_DIC
```

Comment:
Instead of returning in the middle, let's factor out the allocation of the pfDictionary_t to a smaller function and then call it here. We could name it pfCreateDisctionaryStructure(void).

We can also call it  from the new code in pf_save.c.